data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaTopicMetrics.java

```diff
@@ -0,0 +1,109 @@
+        camelCaseMap.put("incoming-byte-rate", "incomingByteRate");
+        camelCaseMap.put("outgoing-byte-rate", "outgoingByteRate");
+        camelCaseMap.put("assigned-partitions", "outgoingByteRate");
+    }
+
+    public void register(KafkaConsumer consumer) {
+        this.consumerMetricsMap.put(consumer, new HashMap<>());
+    }
+
+    private String getCamelCaseName(final String name) {
+        String camelCaseName = camelCaseMap.get(name);
+        if (Objects.isNull(camelCaseName)) {
+            return name;
+        }
+        return camelCaseName;
+    }
+
+    public void update(final KafkaConsumer consumer, final String metricName, Integer metricValue) {
+        synchronized(consumerMetricsMap) {
+            Map<String, Object> cmetrics = consumerMetricsMap.get(consumer);
+            if (cmetrics != null) {
+                cmetrics.put(metricName, (double)metricValue);
+            }
+        }
+    }
+
+    public void update(final KafkaConsumer consumer) {
+        Map<String, Object> cmetrics = null;
+        synchronized(consumerMetricsMap) {
+            cmetrics = consumerMetricsMap.get(consumer);
+        }
+        if (cmetrics == null) {
+            return;
+        }
+        Map<MetricName, ? extends Metric> metrics = consumer.metrics();
+        for (Map.Entry<MetricName, ? extends Metric> entry : metrics.entrySet()) {
+            MetricName metric = entry.getKey();
+            Metric value = entry.getValue();
+            String metricName = metric.name();
+            String metricGroup = metric.group();
+            if ((metricName.contains("consumed")) ||
+                ((!metric.tags().containsKey("partition")) &&
+                     (metricName.equals("records-lag-max") || metricName.equals("records-lead-min"))) ||
+                (metricName.equals("commit-rate") || metricName.equals("join-rate") || metricName.equals("commit-total")) ||
+                (metricName.equals("incoming-byte-rate") || metricName.equals("outgoing-byte-rate"))) {
+                    cmetrics.put(metricName, value.metricValue());
+            }
+        }
+        synchronized (consumerMetricsMap) {
+            long curTime = Instant.now().getEpochSecond();
+            if (curTime - updateTime > metricUpdateInterval) {
+                final Map<String, Double> aggregatedMetrics = new HashMap<>();
+                consumerMetricsMap.forEach((c, metricsMap) -> {
+                    Double value = 0.0;
+                    metricsMap.forEach((metricName, metricValue) -> {
+                        if (metricValue instanceof Double) {
+                            aggregatedMetrics.put(metricName, ((Double)metricValue) + aggregatedMetrics.getOrDefault(metricName, 0.0));
+                        }
+                    });
+                });
+                aggregatedMetrics.forEach((name, value) -> {
+                    pluginMetrics.gauge("topic."+topicName+"."+getCamelCaseName(name), value);
```

Comment:
if we are aggregating for a topic then should this function should be called with list of consumers? may be we can call this from main thread for each topic..

Reply:
It is called for list of consumers, but indirectly.